### DISCUSSION TITLE
Insert vectors of large size will cause "connection reset by peer"

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#16054] [Q&A and General discussion]
Hi, I use cpp client insert 400 million vectors with 64 parallelism in 64 threads. I found that error occurred when I insert 680k vectors in child thread, error shows "Connection reset by peer". But when I cut of my data to smaller size (50k) . It works fine and finishes insertion perfectly. I am confused because my vector dimension is not large( dimension = 32).

---------------------------------
my total data:
400 million * 32 float vector
400 million id (type = int64_t)
these data are divided into 64 groups (child thread), in one group these data are divided into small pieces of size = 680k to insert into milvus.

680k means, 680000 * 32 * 4 / (1024 * 1024) = 83 MB. This is much lower than rpc limitation (500MB).

---------------------------------
my code :

for (int i = 0; i < 64; i++) {
      threads.push_back(std::thread([=, &field_mid_name, &field_embedding_name]() {  
        //每个文件程创建一个新的client链接
        auto client_s = milvus::MilvusClient::Create();
        milvus::ConnectParam connect_param{IP, port};
        auto status = client_s->Connect(connect_param);
        CheckStatus("Failed to connect milvus server:", status);
        for (int k = i * gap; k < gap * (i + 1) && k < size; k++) {
          //xxx....xxxx
          // mid_list size = 680k; emb_list size = 680k
          std::vector<milvus::FieldDataPtr> fields_data{
              std::make_shared<milvus::Int64FieldData>(field_mid_name,
                                                       mid_list),
              std::make_shared<milvus::FloatVecFieldData>(field_embedding_name,
                                                          emb_list)};
          milvus::DmlResults dml_results;
          status = client_s->Insert(collection_name, partition_name,
                                       fields_data, dml_results);
          // Error shows "Connection reset by peer"                              
          //xxx....xxxx
        }
      }));
}

**Comment by xiaojuan8** [comment]
Another problem: when I call Insert() in multi threads with 100k per insertion. After a few minutes before it finishes all data inserting, error occurred showing "Failed to connect all addresses."   50k size per insertion works fine. It is quite confusing. Is there anything special for concurrent insertion to milvus?

**Comment by yhmo** [comment]
Looks like a bug. Could you create an issue so that QA team can verify it?

**Reply by xiaojuan8** [reply]
👌
